documentation/modules/overview/con-kafka-concepts-producers-consumers.adoc

```diff
@@ -0,0 +1,32 @@
+// This module is included in:
+//
+// overview/assembly-kafka-concepts.adoc
+
+// UserStory: Explain the the concepts of producers and consumers
+
+[id="kafka-concepts-producers-consumers_{context}"]
+= Producers and consumers
+//In the title of concept modules, include nouns or noun phrases that are used in the body text. This helps readers and search engines find the information quickly.
+//Do not start the title of concept modules with a verb. See also _Wording of headings_ in _The IBM Style Guide_.
+
+Producers and consumers send and receive messages (publish and subscribe) through brokers.
+Messages comprise an optional _key_ and a _value_ that contains the message data.
+The key is used to identify the message where appropriate.
+
+.Producers and consumers
+
+image:overview/kafka-concepts-producer-consumer.png[Producers and consumers]
+
+Producer:: A producer sends messages to a broker topic to be written to the end offset of a partition.
+Messages are written to partitions by a producer on a round robin basis, or to a specific partition using the message key.
+Consumer:: A consumer subscribes to a topic and reads messages according to topic, partition and offset.
+Consumer group:: Consumer groups are used to share a typically large data stream generated by multiple producers from a given topic.
```

Comment:
I think it might be better to leave it in the list, as it is shown on the graphic as a distinct entity. However, I have extended the explanation:

Consumer group:: Consumer groups are used to share a typically large data stream generated by multiple producers from a given topic.
Consumers are grouped using a `group.id`, allowing messages to be spread across the members.
Consumers within a group do not read data from the same partition, but can receive data from one or more partitions.